app/views/shared/_error_messages.html.erb

```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.size, "error") %>.
```

Comment:
Dùng I18n nhé

2. Phân biệt count, size và length khác nhau thế nào
chỗ này dùng count đã hợp lý chưa

Reply:
Array#length : length trong array sẽ trả về số phần tử ở trong mảng
Array#size: là alias của array#length method. Nó có cùng tính năng và cách thực hiện như array#length.
Array#count: Khác với length và size. array#count có thể sử dụng đi kèm điều kiện để tính toán số phần tử. Có ba cách gọi như sau:
Array#count: trả về số phần tử trong mảng
Array#count n: trả về số phần tử có giá trị = n trong mảng
Array#count{|i| i.even?}: trả về số phần tử trong mảng dựa trên điều kiện.
Khi đối tượng chứa database: ActiveRecord
**#count – collection.count:** tương đương với dùng câu lệnh sql (SELECT COUNT() FROM…) để truy vấn ra số lượng của activerecord mà chúng ta truy vấn. Một lần gọi count thì nó sẽ thực hiện một truy vấn (SELECT COUNT() FROM…) vào database đồng nghĩa với việc count không được lưu trữ trong đối tượng mà chúng ta gọi.
**#length – collection.length:** khi sử dụng length, nó sẽ đếm số lượng mà không sử dụng thêm câu truy vấn nếu đối tượng đó đã được nạp, ngược lại, nó sẽ nạp toàn bộ object của bạn vào bộ nhớ và thực hiện đếm. Khi gọi lenght lần thứ 2 trở đi, lenght sẽ không còn load từ database nữa, mà trả về luôn kết quả(đã được lưu trong bộ nhớ).
**#size – collection.size**: là sự kết hợp của count và length. Khi object đã đuợc load, nó sẽ thực hiện trên chính object đó trong bộ nhớ (không query vào database, tuơng tự lenght ở lần thứ 2 trở đi), còn khi object đó chưa được load, nó sẽ thực hiện câu lệnh (SELECT COUNT(*) FROM…)